core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala

```diff
@@ -97,6 +99,11 @@ class CassandraReadJournal(system: ExtendedActorSystem, cfg: Config)
   private val writePluginConfig = new CassandraJournalConfig(system, system.settings.config.getConfig(writePluginId))
   private val queryPluginConfig = new CassandraReadJournalConfig(cfg, writePluginConfig)
   private val eventAdapters = Persistence(system).adaptersFor(writePluginId)
+  private val eventsByPersistenceIdDeserializer: CassandraJournal.EventDeserializer =
```

Comment:
Why do they have their own?

Reply:
because of the cache, which is looking at the row structure, which might (theoretically) be different for the messages table and the tags view

I should probably add comment about that.

Reply:
Ah good point!